packages/expo-sharing/android/src/main/java/expo/modules/sharing/SharingModule.kt

```diff
@@ -107,19 +83,4 @@ class SharingModule(
       setTypeAndNormalize(mimeType)
       addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
     }
-
-  override fun onActivityResult(activity: Activity, requestCode: Int, resultCode: Int, data: Intent?) {
-    if (requestCode == REQUEST_CODE && pendingPromise != null) {
-      pendingPromise?.resolve(null)
-      pendingPromise = null
-    }
-  }
```

Comment:
Overall, it looks great, but you can't just remove that function and resolve the promise earlier. The time when the promise will be resolved will be different. 

Before:
js calls `shareAsync` -> intent is invoked -> share activity was opened -> user shares something -> execution goes back to the application -> the promise is resolved

Now: 
js calls `shareAsync` ->  intent is invoked -> promise is resolved

In many cases, the flow from the js perspective looks similar, because when the rn app goes to the background (when the shared activity is active), there is a small time window when the app code can still execute. However, it not always may be true.

Reply:
Yeah, I thought this mightn't be right. I'll put it back. So resolving with null is just a way of notifying the js that an action has been completed? You'd only need to do this in situations where it's some kind of callback? If it all completes inside the AsyncFunction body, you can just let the function complete without resolving?

Reply:
Yes, exactly. The `AsyncFunction` will automatically call `promise.resolve(null)` if the closure doesn't take the promise as an argument and the function is complete. In the case where a promise is the last argument of the function, you take responsibility for managing the promise itself. It's helpful when the function is completed, but the rest of the operation happens on a different thread.